lms/resources/_js_config/__init__.py

```diff
@@ -231,20 +232,8 @@ def enable_error_dialog_mode(self, error_code, error_details=None, message=None)
             )
         )
 
-    def enable_dashboard_mode(self, assignment):
-        self._config.update(
-            {
-                "mode": JSConfig.Mode.DASHBOARD,
-                "dashboard": {
-                    "assignment": {"title": assignment.title},
-                    "assignmentStatsApi": {
-                        "path": self._request.route_path(
-                            "api.assignment.stats", id_=assignment.id
-                        ),
-                    },
-                },
-            }
-        )
+    def enable_dashboard_mode(self, config: DashboardConfig):
+        self._config.update({"mode": JSConfig.Mode.DASHBOARD, "dashboard": config})
```

Comment:
We could drop the `mode` config entirely in future, since the frontend already understands it is on the dashboard by looking at the URL.

Reply:
Let's revisit the mode thing later.